### PR TITLE
Finish task

### PR DESCRIPTION
**Pull request by kristjanhollo**

### CODE REVIEW

File: src/main/java/menu/MenuDelivery.java [hunk 1]
```diff
@@ -1,6 +1,99 @@
...
+import util.DBUtil;
+
+import java.util.List;
+import java.util.Scanner;
+
 public class MenuDelivery {
 
+    RepositoryDelivery repositoryDelivery = new RepositoryDelivery();
+
+    public MenuDelivery() {
```
Comment (infodi) [inline src/main/java/menu/MenuDelivery.java:15]: move 'new RepositoryDelivery to inside the constructor

Comment (kristjanhollo) [reply src/main/java/menu/MenuDelivery.java:15]: Done

File: src/main/java/menu/MenuDelivery.java [hunk 2]
```diff
@@ -1,6 +1,99 @@
...
+            System.out.println(result);
+        }
+    }
+
+    private void listTotalDeliveryTime(Scanner input) {
+        List<Delivery> list = repositoryDelivery.listAllDeliveryTimes();
+        if(list.size() > 0) {
+            System.out.println("\nTotal delivery time:");
+            int sum = 0;
+            for(Delivery delivery : list) {
```
Comment (infodi) [inline src/main/java/menu/MenuDelivery.java:89]: if you have on million of delivery, will iterate a 'loop for' for a million of rows? Not good solution.
you can solve that using group by on repository. 

check the example on RepositoryCustomer line 46

File: src/main/java/menu/MenuOptions.java [hunk 3]
```diff
@@ -21,7 +21,7 @@ private int menuOptions(Scanner input) {
         System.out.println();
         System.out.println("1: Sub Menu Customer");
         System.out.println("2: Sub Menu Restaurant");
-        System.out.println("3: ");
+        System.out.println("3: Sub Menu Delivery");
```
Comment (infodi) [inline src/main/java/menu/MenuOptions.java]: please update your master and then update your branch, there's some updates on this class

File: src/main/java/persistence/RepositoryDelivery.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    private EntityManager em;
+
+    public RepositoryDelivery() {
+        em = DBUtil.getEntityManager();
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Delivery> listAllDeliveryTimes() {
+        return em.createQuery("FROM Delivery ")
```
Comment (infodi) [inline src/main/java/persistence/RepositoryDelivery.java:22]: here need to use 'group by' function.
Checking deliver time should be a date option, like: 23-05-2021:10:20
For the task can keep only the date 23-05-2021
So, Your method should return something like:

date               total
23-05-2021    23
24-05-2021    104

File: src/main/java/menu/MenuDelivery.java (hunk 2 above)
Comment (infodi) [reply src/main/java/menu/MenuDelivery.java:89]: You add three methods the task ask only one method :)

Comment (kristjanhollo) [reply src/main/java/menu/MenuDelivery.java]: Simplified solution

File: src/main/java/menu/MenuOptions.java (hunk 3 above)
Comment (kristjanhollo) [reply src/main/java/menu/MenuOptions.java]: Should be correct

**Review by lecaryna — APPROVED** [review]

**Pull request merged**